File: crates/hive-api/src/routes/workspace.rs
use axum::extract::State;
use axum::http::HeaderMap;
use axum::routing::get;
use axum::{Json, Router};
use chrono::Utc;
use hive_type::id::WorkspaceId;
use hive_type::Workspace;
use serde::Deserialize;

use crate::error::ApiError;
use crate::routes::current_user;
use crate::AppState;

#[derive(Deserialize)]
struct CreateWorkspaceRequest {
    name: String,
    slug: String,
    description: Option<String>,
    avatar: Option<String>,
    instructions: Option<String>,
    repositories: Option<Vec<String>>,
    auto_pr_review: Option<bool>,
    auto_issue_assign: Option<bool>,
}

pub fn router() -> Router<AppState> {
    Router::new()
        .route("/", get(list_workspaces).post(create_workspace))
}

async fn create_workspace(
    State(state): State<AppState>,
    headers: HeaderMap,
    Json(req): Json<CreateWorkspaceRequest>,
) -> Result<Json<Workspace>, ApiError> {
    let user = current_user(&state, &headers).await?;
    let now = Utc::now();
    let workspace = Workspace {
        id: WorkspaceId::new(),
        creator_id: user.id,
        name: req.name,
        slug: req.slug,
        description: req.description,
        avatar: req.avatar,
        instructions: req.instructions,
        repositories: req.repositories.unwrap_or_default(),
        auto_pr_review: req.auto_pr_review.unwrap_or(false),
        auto_issue_assign: req.auto_issue_assign.unwrap_or(false),
        created_at: now,
        updated_at: now,
    };
    let created = hive_database::repository::create_workspace(&state.db, workspace).await?;
    Ok(Json(created))
}

async fn list_workspaces(
    State(state): State<AppState>,
    headers: HeaderMap,
) -> Result<Json<Vec<Workspace>>, ApiError> {
    let user = current_user(&state, &headers).await?;
    let workspaces =
        hive_database::repository::list_workspaces_for_user(&state.db, user.id).await?;
    Ok(Json(workspaces))
}

File: apps/web/src/services/workspace.ts
import type { Workspace } from '../types';

export interface CreateWorkspaceRequest {
  name: string;
  slug: string;
  description?: string | null;
  avatar?: string | null;
  instructions?: string | null;
  repositories?: string[];
  auto_pr_review?: boolean;
  auto_issue_assign?: boolean;
}

export const fetchWorkspaces = async (): Promise<Workspace[]> => {
  const res = await fetch('/api/workspaces');
  if (!res.ok) return [];
  return res.json();
};

export const createWorkspace = async (data: CreateWorkspaceRequest): Promise<Workspace> => {
  const res = await fetch('/api/workspaces', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(data),
  });
  if (!res.ok) {
    const err = await res.json().catch(() => ({ error: '创建失败' }));
    throw new Error(err.error || '创建空间失败');
  }
  return res.json();
};
